compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -421,6 +422,8 @@ private void defineFunction(DataInputStream dataInStream) throws IOException {
             if (attachedType.tag == TypeTags.OBJECT || attachedType.tag == TypeTags.RECORD) {
                 scopeToDefine = attachedType.tsymbol.scope;
                 if (isResourceFunction) {
+                    // If it is a resource function, attached type should be an object
+                    BObjectTypeSymbol objectTypeSymbol = (BObjectTypeSymbol) attachedType.tsymbol;
```

Comment:
It is better to define this just before the place where it uses.

Reply:
Updated in [c169d3b](https://github.com/ballerina-platform/ballerina-lang/pull/38163/commits/c169d3b53e8b4a69574adaa63da5fa0ef9dbc976)